### PR TITLE
Get balances for multiple addresses

### PR DESCRIPTION
**Pull request by jojobyte**
`getInstantBalances` uses the POST Endpoint from <https://github.com/dashpay/insight-api#unspent-outputs-for-multiple-addresses> to get balances for multiple addresses.

```js
let addresses = [
  'XmaroZwvCKjsYQQVdCQP2BuWnnMGxWLCGJ',
  'XmCyQ6qARLWXap74QubFMunngoiiA1QgCL',
  'XaxrcNUS6MrAfsvXNF2s24eChFVKabU6gP',
]
let balances = await dashsight.getInstantBalances(addresses)

console.log(balances)
```
Returns all addresses with UTXO's

```json
[
  {
    "addrStr": "XmCyQ6qARLWXap74QubFMunngoiiA1QgCL",
    "balance": 0.00099809,
    "balanceSat": 99809,
    "_utxoCount": 1,
    "_utxoAmounts": [
      99809
    ]
  }
]
```

`getMultiCoreUtxos` uses the POST Endpoint from <https://github.com/dashpay/insight-api#unspent-outputs-for-multiple-addresses> to get balances for multiple addresses and parse with `toCoreUtxos` to be used in place of `getCoreUtxos`

```js
let addresses = [
  'XmaroZwvCKjsYQQVdCQP2BuWnnMGxWLCGJ',
  'XmCyQ6qARLWXap74QubFMunngoiiA1QgCL',
  'XaxrcNUS6MrAfsvXNF2s24eChFVKabU6gP',
]
let utxos = await dashsight.getMultiCoreUtxos(addresses)

console.log(utxos)
```
Returns all addresses with UTXO's

```json
[
  {
    "address": "XmaroZwvCKjsYQQVdCQP2BuWnnMGxWLCGJ",
    "outputIndex": 0,
    "satoshis": 99809,
    "script": "00000000000000000000000000000000000000000000000000",
    "txId": "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
  }
]
```

### CODE REVIEW

File: dashsight.js [hunk 1]
```diff
@@ -118,6 +155,39 @@
       return utxos;
     };
 
+    /** @type {(addresses: string | string[]) => String} */
+    insight.joinAddrs = function (addresses) {
```
Comment (coolaj86) [inline dashsight.js:159]: I would leave this an internal implementation detail:
`insight._joinAddrs`

File: dashsight.js [hunk 2]
```diff
@@ -221,8 +297,8 @@
       txs.forEach(function (tx, i) {
         //let fee = tx.valueIn - tx.valueOut;
         // consumed as an input
-        tx.vin.forEach(removeSpentOutputs);
         tx.vout.forEach(addUnspentOutputs);
+        tx.vin.forEach(removeSpentOutputs);
```
Comment (coolaj86) [inline dashsight.js:301]: Why was this moved?

**Review by coolaj86 — APPROVED** [review]

File: dashsight.js [hunk 3]
```diff
@@ -24,7 +24,6 @@
   /** @typedef {import('./').InsightTxVin} InsightTxVin */
   /** @typedef {import('./').InsightTxVout} InsightTxVout */
   /** @typedef {import('./').InsightUtxo} InsightUtxo */
-  // /** @typedef {import('./').InsightAddrsTxResponse} InsightAddrsTxResponse */
```
Comment (coolaj86) [inline dashsight.js]: Some of these may be reexports to be used in other libraries.

File: dashsight.js (hunk 2 above)
Comment (jojobyte) [reply dashsight.js:301]: `removeSpentOutputs` in `tx.vin.forEach(removeSpentOutputs)` is trying to loop over the data added by `tx.vout.forEach(addUnspentOutputs)`, it made sense to me to have `addUnspentOutputs` start first.

File: dashsight.js (hunk 3 above)
Comment (jojobyte) [reply dashsight.js]: All of these were things I committed while testing this PR and I removed after validating they were unnecessary.
